happiness-backend/api/routes/user.py

```diff
@@ -202,80 +200,71 @@ def change_user_info(req, headers):
             return failure_response("Provided data already exists", 400)
 
         current_user.email = new_email
-        db.session.commit()
-        return current_user
     elif data_type == "password":
-        # Changes a user's password.
-        try:
-            current_user.change_password(req.get("data"), headers.get("password_key"))
-            db.session.commit()
-            return current_user, {
-                'Password-Key': current_user.derive_pwd_key(req.get('data'))
-            }
-        except Exception as e:
-            print(e)
-            return failure_response('Invalid password key.', 400)
+        # Changes a user's password, which requires their old password and a new one.
+        old_password, new_password = req.get("data"), req.get("data2")
+        if not current_user.verify_password(old_password):
+            return failure_response("Incorrect Password", 401)
+
+        current_user.change_password(old_password, new_password)
     elif data_type == "key_recovery_phrase":
-        # Adds/Changes key recovery phrase to prevent loss of encrypted data on password reset.
-        try:
-            current_user.add_key_recovery(req.get("data"), headers.get("password_key"))
-            db.session.commit()
-            return current_user
-        except Exception as e:
-            print(e)
-            return failure_response('Invalid password key.', 400)
+        # Adds/Changes key recovery phrase to prevent loss of encrypted data on password reset,
```

Comment:
Could you add a bit more documentation about how data isn't lost here? I'm curious to see how this works, are the journals re-encrypted using a new password key or something?

Reply:
The user_key is encrypted with the user's password. The recovery_key is basically just the user_key encrypted with the recovery phrase. currently there are no length requirements to it but maybe we should add them in the future to make it more secure